pkg/core/native/native_test/cryptolib_test.go

```diff
@@ -0,0 +1,268 @@
+}
+
+func TestCryptolib_TestBls12381Pairing_Compat(t *testing.T) {
+	c := newCryptolibClient(t)
+
+	script := io.NewBufBinWriter()
+	emit.AppCall(script.BinWriter, c.Hash, "bls12381Deserialize", callflag.All, g2)
+	emit.AppCall(script.BinWriter, c.Hash, "bls12381Deserialize", callflag.All, g1)
+	emit.Opcodes(script.BinWriter, opcode.PUSH2, opcode.PACK)
+	emit.AppCallNoArgs(script.BinWriter, c.Hash, "bls12381Pairing", callflag.All)
+
+	stack, err := c.TestInvokeScript(t, script.Bytes(), c.Signers)
+	require.NoError(t, err)
+	require.Equal(t, 1, stack.Len())
+	itm := stack.Pop().Item()
+	require.Equal(t, stackitem.InteropT, itm.Type())
+	actual, ok := itm.(*stackitem.Interop).Value().(*bls12381.GT)
+	require.True(t, ok)
+	arr := actual.Bytes()
+	// Expected value is taken from the reference test.
+	require.Equal(t, strings.ToLower("0F41E58663BF08CF068672CBD01A7EC73BACA4D72CA93544DEFF686BFD6DF543D48EAA24AFE47E1EFDE449383B67663104C581234D086A9902249B64728FFD21A189E87935A954051C7CDBA7B3872629A4FAFC05066245CB9108F0242D0FE3EF03350F55A7AEFCD3C31B4FCB6CE5771CC6A0E9786AB5973320C806AD360829107BA810C5A09FFDD9BE2291A0C25A99A211B8B424CD48BF38FCEF68083B0B0EC5C81A93B330EE1A677D0D15FF7B984E8978EF48881E32FAC91B93B47333E2BA5706FBA23EB7C5AF0D9F80940CA771B6FFD5857BAAF222EB95A7D2809D61BFE02E1BFD1B68FF02F0B8102AE1C2D5D5AB1A19F26337D205FB469CD6BD15C3D5A04DC88784FBB3D0B2DBDEA54D43B2B73F2CBB12D58386A8703E0F948226E47EE89D018107154F25A764BD3C79937A45B84546DA634B8F6BE14A8061E55CCEBA478B23F7DACAA35C8CA78BEAE9624045B4B601B2F522473D171391125BA84DC4007CFBF2F8DA752F7C74185203FCCA589AC719C34DFFBBAAD8431DAD1C1FB597AAA5193502B86EDB8857C273FA075A50512937E0794E1E65A7617C90D8BD66065B1FFFE51D7A579973B1315021EC3C19934F1368BB445C7C2D209703F239689CE34C0378A68E72A6B3B216DA0E22A5031B54DDFF57309396B38C881C4C849EC23E87089A1C5B46E5110B86750EC6A532348868A84045483C92B7AF5AF689452EAFABF1A8943E50439F1D59882A98EAA0170F1250EBD871FC0A92A7B2D83168D0D727272D441BEFA15C503DD8E90CE98DB3E7B6D194F60839C508A84305AACA1789B6"),
+		hex.EncodeToString(arr[:]))
+}
+
+func TestCryptolib_TestBls12381Equal_Compat(t *testing.T) {
+	c := newCryptolibClient(t)
+
+	script := io.NewBufBinWriter()
+	emit.AppCall(script.BinWriter, c.Hash, "bls12381Deserialize", callflag.All, g1)
+	emit.AppCall(script.BinWriter, c.Hash, "bls12381Deserialize", callflag.All, g1)
+	emit.Opcodes(script.BinWriter, opcode.PUSH2, opcode.PACK)
+	emit.AppCallNoArgs(script.BinWriter, c.Hash, "bls12381Equal", callflag.All)
+
+	stack, err := c.TestInvokeScript(t, script.Bytes(), c.Signers)
+	require.NoError(t, err)
+	require.Equal(t, 1, stack.Len())
+	itm := stack.Pop().Item()
+	require.Equal(t, stackitem.BooleanT, itm.Type())
+	require.True(t, itm.Value().(bool))
+}
+
+// Data used for this test are generated by C# library.
+func TestCryptolib_TestBls12381Mul_CompatCustom(t *testing.T) {
```

Comment:
@roman-khimov, do we need more tests like that for other operations/points? This test uses exact points represented as hex-encoded strings that were generated randomply. The `expected` result is taken "as is" directly from the C# operation result.

Reply:
It'd be nice to have _exactly the same_ data in C# tests. This way we'll ensure that those number really are correct and that they won't break in the future.